Cli.Commands.Abstractions/CliCommandMap.cs

```diff
@@ -0,0 +1,22 @@
+using Cli.Commands.Abstractions.Extensions;
+using Cli.Instructions.Abstractions;
+
+namespace Cli.Commands.Abstractions;
+
+public class CliCommandMap
+{
+    private readonly List<string> _nextCommands = new();
```

Comment:
Might be able to just inherit the list and add the method to extend list functionality.